src/main/java/ch/cern/eam/wshub/core/services/equipment/impl/NonConformityObservationServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+import net.datastream.schemas.mp_functions.mp3443_001.MP3443_SyncNonconformityObservation_001;
+import net.datastream.schemas.mp_functions.mp3444_001.MP3444_DeleteNonconformityObservation_001;
+import net.datastream.schemas.mp_functions.mp3445_001.MP3445_GetNonconformityObservation_001;
+import net.datastream.schemas.mp_results.mp3402_001.MP3402_CreateNonconformityObservation_001_Result;
+import net.datastream.schemas.mp_results.mp3442_001.MP3442_AddNonconformityObservation_001_Result;
+import net.datastream.schemas.mp_results.mp3443_001.MP3443_SyncNonconformityObservation_001_Result;
+import net.datastream.schemas.mp_results.mp3444_001.MP3444_DeleteNonconformityObservation_001_Result;
+import net.datastream.schemas.mp_results.mp3445_001.MP3445_GetNonconformityObservation_001_Result;
+import net.datastream.wsdls.inforws.InforWebServicesPT;
+
+import java.math.BigDecimal;
+
+import static ch.cern.eam.wshub.core.tools.DataTypeTools.isEmpty;
+
+public class NonConformityObservationServiceImpl implements NonConformityObservationService {
+
+    private ApplicationData applicationData;
+    private Tools tools;
+    private InforWebServicesPT inforws;
+    private NonconformityService  nonconformityService;
+
+    public NonConformityObservationServiceImpl(ApplicationData applicationData, Tools tools, InforWebServicesPT inforWebServicesToolkitClient) {
+        this.applicationData = applicationData;
+        this.tools = tools;
+        this.inforws = inforWebServicesToolkitClient;
+        this.nonconformityService = new NonconformityServiceImpl(applicationData, tools, inforws);
+    }
+
+    @Override
+    public String createNonConformityObservation(InforContext context, NonConformityObservation nonConformityObservation) throws InforException {
+        if(nonConformityObservation.getJobNum() != null) {
+            MP3442_AddNonconformityObservation_001 addNonconformityObservation = new MP3442_AddNonconformityObservation_001();
+            NonconformityObservation nonconformityObservation =  tools.getInforFieldTools().transformWSHubObject(
+                    createDefaultNonConformityObservation(), nonConformityObservation, context);
+
+
+            addNonconformityObservation.setNonconformityObservation(nonconformityObservation);
+            MP3442_AddNonconformityObservation_001_Result result = tools.performInforOperation(context, inforws::addNonconformityObservationOp, addNonconformityObservation);
+            return result.getResultData().getNONCONFORMITYOBSERVATIONID().getOBSERVATIONPK();
+
+        } else {
+            MP3402_CreateNonconformityObservation_001 addNonconformityObservation = new MP3402_CreateNonconformityObservation_001();
+            NonconformityObservation nonconformityObservation =  tools.getInforFieldTools().transformWSHubObject(
+                    new NonconformityObservation(), nonConformityObservation, context);
+            STANDARDENTITYID_Type typeId = new STANDARDENTITYID_Type();
```

Comment:
Id Type are different so we need to map it

Reply:
Can't we use the same multiple xpaths strategy?